lib/core/managementClient.ts

```diff
@@ -532,6 +556,224 @@ export class ManagementClient extends LinkEntity {
+        this._context.namespace.connectionId);
+      await defaultLock.acquire(this.managementLock, () => { return this._init(); });
+
+      const response = await this._mgmtReqResLink!.sendRequest(request);
+      if (response.application_properties!.statusCode === 204 || !response.body || !Array.isArray(response.body.rules)) {
+        return [];
+      }
+
+      // Reference: https://docs.microsoft.com/en-us/azure/service-bus-messaging/service-bus-amqp-request-response#response-11
+      const result: { "rule-description": Typed }[] = response.body.rules || [];
+      const rules: RuleDescription[] = [];
+      result.forEach((x) => {
+        const ruleDescriptor = x['rule-description'];
+
+        // We use the first three elements of the `ruleDescriptor.value` to get filter, action, name
+        if (!ruleDescriptor
+          || !ruleDescriptor.descriptor
+          || ruleDescriptor.descriptor.value !== descriptorCodes.ruleDescriptionList
+          || !Array.isArray(ruleDescriptor.value)
+          || ruleDescriptor.value.length < 3) {
+          return;
+        }
+
+        const filtersRawData: Typed = ruleDescriptor.value[0];
+        const actionsRawData: Typed = ruleDescriptor.value[1];
+        const rule: RuleDescription = {
+          name: ruleDescriptor.value[2].value
+        };
+
+        switch (filtersRawData.descriptor.value) {
+          case descriptorCodes.trueFilterList:
+            rule.filter = {
+              expression: "1=1"
+            };
+            break;
+          case descriptorCodes.falseFilterList:
+            rule.filter = {
+              expression: "1=0"
+            };
+            break;
+          case descriptorCodes.sqlFilterList:
+            if (Array.isArray(filtersRawData.value) && filtersRawData.value.length === 2) {
+              rule.filter = {
+                expression: this._safelyGetTypedValue(filtersRawData.value[0])
+              };
+            }
+            break;
+          case descriptorCodes.correlationFilterList:
+            if (Array.isArray(filtersRawData.value) && filtersRawData.value.length === 9) {
+              rule.filter = {
+                correlationId: this._safelyGetTypedValue(filtersRawData.value[0]),
+                messageId: this._safelyGetTypedValue(filtersRawData.value[1]),
+                to: this._safelyGetTypedValue(filtersRawData.value[2]),
+                replyTo: this._safelyGetTypedValue(filtersRawData.value[3]),
+                label: this._safelyGetTypedValue(filtersRawData.value[4]),
+                sessionId: this._safelyGetTypedValue(filtersRawData.value[5]),
+                replyToSessionId: this._safelyGetTypedValue(filtersRawData.value[6]),
+                contentType: this._safelyGetTypedValue(filtersRawData.value[7]),
+                userProperties: this._safelyGetTypedValue(filtersRawData.value[8]),
+              };
+            }
+            break;
+          default:
+            break;
```

Comment:
- We should definitely log `filtersRawData.descriptor.value` when we land in the default case with a statement indicating that we have landed in the default case. It can be useful when asking the customer to send debug logs.
- Currently we are dropping the rule description deserialization if it lands in the default case. Should we throw an error instead or may be put the raw value as is?

Reply:
Logged the case when the descriptor code is not among the expected values.
In such cases, do you use log.mgmt or log.error?

> Currently we are dropping the rule description deserialization if it lands in the default case. Should we throw an error instead or may be put the raw value as is?

The rule description continues, only the filter is dropped.